update v3 scheduler section (#33)
ankitects/faqs: src/the-2021-scheduler.md

> 

Reviewer: This is a long-time peeve of mine, so as long as we're here ...

Since "v3" is the name that stuck, and it's the name in the left nav bar, can we change the page headline to "The Anki v3 scheduler"? That won't turn "the-2021-scheduler.html" into a deadlink, will it? Preserving links to this page is much more important!

Author: Below the header I'd still like "2021 scheduler" being mentioned. A little bit of Anki trivia won't hurt. This is something we would also want to do for "2.1 scheduler" (in this case, the manual will also need correcting).

Reviewer: I'm totally on board with leaving the sentence ("The 2021 scheduler ("v3") is an update to ...") as-is. Just the title.

As of Anki/AnkiMobile 23.10, and AnkiDroid 2.17, the v3 scheduler is the default and only option.

> On earlier versions, the scheduler can be changed from the preferences screen.

Author: The preposition *from* felt more natural here. Although, I agree I could've kept the *in* in there.

As for the word *old*, I think it comes with a negative connotation oftentimes. Using *earlier* is neutral and is more clear. One might think, because they downloaded Anki a few months ago, it's probably not "old".

Because the v3 scheduler uses a different approach to gathering and sorting
cards, a v2 and v3 client may show a different number of due cards on a given
day, and may show them in a different order.

## Changes

### Undo

The v3 scheduler uses Anki's new undo infrastructure: you can answer a card,
bury another card, answer a different card, then undo each action in turn. Previous schedulers handled undo for review and non-review actions separately, so a non-review action
would clear the review history, and vice versa.

### Daily limits

> The new card limit in v3 scheduler is capped by the review limit by default, so that new card introduction

Reviewer: ```suggestion
By default, the new card limit in the v3 scheduler is capped by the review limit, so that new card introduction
```

Author: Is this any better? I don't want to put too much attention here so maybe we can just copy the tooltips:

```suggestion
By default, the review limit in v3 scheduler also applies to new cards, so that new card introduction
```

Reviewer: ```suggestion
By default, the review limit in the v3 scheduler also applies to new cards, so that new card introduction
```

is reduced/paused when you have a backlog of reviews. For example, if you have limits of

> 200 review card and 20 new cards, and 190 review cards are due, only 10 new

Reviewer: ```suggestion
200 review cards and 20 new cards, and 190 review cards are due, only 10 new
```

cards will be introduced. This prevents the backlog from getting
worse.

If you find this happening, the recommended solution is to increase your review limit, and work through
the backlog before you add more new cards. If you're sure you want to add more new cards
even though you have a backlog, you can enable the "New cards ignore review limit" option
in deck options.

> Each deck's limits affects the number of cards that will be gathered from that deck

Reviewer: ```suggestion
The limits of each deck control the number of cards that will be gathered from that deck
```

The v3 scheduler uses Anki's new undo infrastructure: you can answer a card,
bury another card, answer a different card, then undo each action in turn. Previous schedulers handled undo for review and non-review actions separately, so a non-review action

> would clear the review history, or the other way around.

Reviewer: I don't think "or the other way around" makes sense here. It sounds like you're saying, "the review history would clear a non-review action." "Vice versa" is better, but if you don't want to keep that -- 

- make it a "so for example" and drop that phrase from the end. 
- make it generic -- "so performing an action for one would clear the undo history for the other."

even though you have a backlog, you can enable the "New cards ignore review limit" option
in deck options.

> In a deck tree, the limits of each deck will control the number of cards that gathered from that deck

Reviewer: ```suggestion
The limits of each deck will control the number of cards that are gathered from that deck
```
I think with "deck and its subdecks," you don't need to use "in a deck tree."

and its subdecks. Limits are applied from the deck you select, so if you select

> a child deck, its parents' limits will not apply. For example, if the
> following are the limits:

Reviewer: I prefer the original wording and don't think this needs to change, but if it changes, "are the limits" doesn't sound right to me. Perhaps this instead? 
```suggestion
a child deck, its parents' limits will not apply. For example, if the
following limits are set:
```

Then:

- If you click "Grandchild3", you'll get up to 200 cards.
- If you click "Grandchild2", you'll get up to 5 cards.
- If you click "Grandchild1", you'll get up to 50 cards.
- If you click "Child", you'll get up to 30 cards from the "Child" deck and its
  subdecks. No more than 5 cards will be taken from "Grandchild2".
- If you click "Parent", you'll get up to 100 cards, with a maximum of 30

>   coming from "Child" and its subdecks.

Reviewer: I think the quotation marks just clutter up this example, and make it hard to visually follow (especially up against all of those commas). If you don't like "click X" -- and I see you already changed it from "click on X" (which seems fine) -- try "select X" -- which uses the same language as the description above the example.

Then:

- If you click Grandchild3, you'll get up to 200 cards.
- If you click Grandchild2, you'll get up to 5 cards.
- If you click Grandchild1, you'll get up to 50 cards.
- If you click Child, you'll get up to 30 cards from the Child deck and its
  subdecks. No more than 5 cards will be taken from Grandchild2.
- If you click Parent, you'll get up to 100 cards, with a maximum of 30
  coming from Child and its subdecks.

> In earlier Anki releases, v3 scheduler did not respect intermediate limits, so when clicking on Parent, the limits of Child did not influence how

Reviewer: ```suggestion
In earlier Anki releases, the v3 scheduler did not respect intermediate limits, so when clicking on Parent, the limits of Child did not influence how
```

reviews are presented in. New cards can be mixed from multiple decks, and
reviews can optionally be ordered by interval or subdeck.

> When burying is disabled, with v3 scheduler it's possible to control whether siblings are

Reviewer: ```suggestion
When burying is disabled, with the v3 scheduler it's possible to control whether siblings are
```

### Burying

When burying is enabled, cards are now excluded from the card counts at the start of
a study session. Previously if you had 10 forward and 10 reversed cards, the
counts would start at 20 and jump down as you review, but with v3 they'll start directly
at 10.

> Because burying is done when you click on a deck, the counts you see on the deck

Author: It wasn't very clear to me what is "exclusion" so I thought it's burying. Can you rewrite this?

> Because burying is done when you click on a deck, the counts you see on the deck

Reviewer 2: ```suggestion
Because siblings are skipped when you click on a deck, the counts you see on the deck
```

list will differ from the ones you see when you click on a deck. The overview screen

> will point out the number of cards that will be buried in grey.

Reviewer 2: ```suggestion
will point out the number of cards that were waiting but have been skipped.
```

### Interday learning

> Interday (1+ day) learning cards are now subject to the review limit. When

Author: we already defined this, is there a need for this? bracketed text reduce readability.

> Interday (1+ day) learning cards are now subject to the review limit. When

Reviewer: Correct. I'm suggesting we remove the inaccurate parenthetical.

determining what fits within the limit, Anki gathers interday learning cards
first, then review cards, and finally new cards.

### Filtered decks

Filtered decks with rescheduling disabled show 4 answer buttons. You can configure a time delay for each button. Setting this delay to 0 makes a button remove the card from the deck.

In earlier versions, a single delay was provided for Again. Hard used 1.5x the provided delay, whereas Good used 2x of the delay.

> Easy removed the card.

Author: See my latest commit

> Easy removed the card.

Reviewer: Looks good! But I don't know how that "whereas" snuck through in my suggestion. I distinctly remember cutting it. I don't think we need it, and it's a "too fancy" sort of word.

```

Because this is implemented in JavaScript, it is not limited to the computer
version. AnkiMobile and AnkiDroid both support it as well, and AnkiWeb may also
support it in the future. This will allow advanced users to make 
adjustments to the standard scheduling behaviour, that apply on all platforms.

> The various scheduling states are described in [Scheduling States](https://github.com/ankitects/anki/blob/main/proto/anki/scheduler.proto).

Reviewer: ```suggestion
The various scheduling states are described in [SchedulingStates](https://github.com/ankitects/anki/blob/main/proto/anki/scheduler.proto).
```
That's what it's called in the code -- no space.

Author: Typo on my part.
